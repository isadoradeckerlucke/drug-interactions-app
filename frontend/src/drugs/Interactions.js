import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import SearchForm from "./SearchForm";
import InteractionList from "./InteractionList";
import { resetLoading } from "../actions/resetLoading";
import LoadingPage from "../LoadingPage";
import { setSearchTerm } from "../actions/searchTerm";

function Interactions() {
  const interactions = useSelector((st) => st.drugs.interactions);
  const loading = useSelector((st) => st.loading.loading);
  const searchTerm = useSelector((st) => st.searchTerm.searchTerm);
  const dispatch = useDispatch();

  useEffect(() => {
    if (interactions) {
      dispatch(resetLoading());
      dispatch(setSearchTerm(""));
    }
  }, [dispatch, interactions]);

  return (
    <div>
      <SearchForm />

      {loading ? (
        <LoadingPage searchTerm={searchTerm}></LoadingPage>
      ) : (
        <p id="error-holder"></p>
      )}

      {interactions ? (
        Object.keys(interactions).length !== 0 ? (
          <InteractionList interactions={interactions} />
        ) : (
          <p>sorry, no results were found</p>
        )
      ) : (
        <p></p>
      )}
    </div>
  );
}

export default Interactions;
