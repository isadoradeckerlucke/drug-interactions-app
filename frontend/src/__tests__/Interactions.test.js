import React from "react";
import { render } from "@testing-library/react";
import Interactions from "../drugs/Interactions";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("interactions page works as expected", function () {
  const currentUserTest = {
    isLoggedIn: true,
    currentUser: {
      username: "testuser",
      email: "testuser@gmail.com",
      savedDrugs: [],
    },
  };

  const initialState = {
    drugs: { interactions: {} },
    loading: { loading: false },
    searchTerm: { searchTerm: "adderall" },
    currentUser: currentUserTest,
  };

  const loadingState = {
    drugs: { interactions: {} },
    loading: { loading: true },
    searchTerm: { searchTerm: "adderall" },
    currentUser: currentUserTest,
  };

  const notFoundState = {
    drugs: { interactions: {} },
    loading: { loading: false },
    searchTerm: { searchTerm: "hello" },
    currentUser: currentUserTest,
  };

  const foundState = {
    drugs: {
      interactions: {
        "00": {
          comment: "test drug is resolved to test drug",
          description: "test description",
          name1: "drug1test",
          name2: "drug2test",
          severity: "high",
        },
        "01": {
          comment: "test drug is resolved to test drug",
          description: "test description",
          name1: "drug1test2",
          name2: "drug2test2",
          severity: "N/A",
        },
      },
    },
    loading: { loading: false },
    searchTerm: { searchTerm: "adderall" },
    currentUser: currentUserTest,
  };

  const mockStore = configureStore();
  let store;

  /** smoke test */
  it("renders without crashing", function () {
    store = mockStore(initialState);
    render(
      <Provider store={store}>
        <Interactions />
      </Provider>
    );
  });

  /** snapshot test */
  it("matches initial snapshot", function () {
    store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <Interactions />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when loading", function () {
    store = mockStore(loadingState);

    const { asFragment } = render(
      <Provider store={store}>
        <Interactions />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot no interactions found", function () {
    store = mockStore(notFoundState);

    const { asFragment } = render(
      <Provider store={store}>
        <Interactions />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot interactions found", function () {
    store = mockStore(foundState);

    const { asFragment } = render(
      <Provider store={store}>
        <Interactions />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
