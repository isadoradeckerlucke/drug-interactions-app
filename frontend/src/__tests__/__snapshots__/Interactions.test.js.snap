// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interactions page works as expected matches initial snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        search for high severity interactions with a single drug, or get all interactions between multiple drugs.
      </h3>
      <form
        class=""
      >
        <label
          class="form-label"
          for="drugName"
        >
          search interactions:
        </label>
        <input
          class="form-control"
          name="drugName"
          placeholder="drug name(s)"
          value="adderall"
        />
        <br />
        <div
          class="form-group"
        >
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="formBasicCheckbox"
              name="saveBool"
              type="checkbox"
              value="false"
            />
            <label
              class="form-check-label"
              for="formBasicCheckbox"
              title=""
            >
              add searched drugs to saved list
            </label>
          </div>
        </div>
        <small
          class="form-text text-muted"
          id="searchHelpBlock"
        >
          please separate multiple drug names by commas
        </small>
        <br />
        <button
          class="btn btn-lg btn-dark btn btn-primary"
          type="submit"
        >
          search
        </button>
        <p
          id="error-holder"
        />
      </form>
      <br />
    </div>
    <p
      id="error-holder"
    />
    <p>
      sorry, no results were found
    </p>
  </div>
</DocumentFragment>
`;

exports[`interactions page works as expected matches snapshot interactions found 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        search for high severity interactions with a single drug, or get all interactions between multiple drugs.
      </h3>
      <form
        class=""
      >
        <label
          class="form-label"
          for="drugName"
        >
          search interactions:
        </label>
        <input
          class="form-control"
          name="drugName"
          placeholder="drug name(s)"
          value="adderall"
        />
        <br />
        <div
          class="form-group"
        >
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="formBasicCheckbox"
              name="saveBool"
              type="checkbox"
              value="false"
            />
            <label
              class="form-check-label"
              for="formBasicCheckbox"
              title=""
            >
              add searched drugs to saved list
            </label>
          </div>
        </div>
        <small
          class="form-text text-muted"
          id="searchHelpBlock"
        >
          please separate multiple drug names by commas
        </small>
        <br />
        <button
          class="btn btn-lg btn-dark btn btn-primary"
          type="submit"
        >
          search
        </button>
        <p
          id="error-holder"
        />
      </form>
      <br />
    </div>
    <p
      id="error-holder"
    />
    <div>
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                drug 1
              </th>
              <th>
                drug 2
              </th>
              <th>
                description
              </th>
              <th>
                severity
              </th>
              <th>
                comment
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                drug1test
              </td>
              <td>
                drug2test
              </td>
              <td>
                test description
              </td>
              <td>
                high
              </td>
              <td>
                test drug is resolved to test drug
              </td>
            </tr>
            <tr>
              <td>
                drug1test2
              </td>
              <td>
                drug2test2
              </td>
              <td>
                test description
              </td>
              <td>
                N/A
              </td>
              <td>
                test drug is resolved to test drug
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`interactions page works as expected matches snapshot no interactions found 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        search for high severity interactions with a single drug, or get all interactions between multiple drugs.
      </h3>
      <form
        class=""
      >
        <label
          class="form-label"
          for="drugName"
        >
          search interactions:
        </label>
        <input
          class="form-control"
          name="drugName"
          placeholder="drug name(s)"
          value="hello"
        />
        <br />
        <div
          class="form-group"
        >
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="formBasicCheckbox"
              name="saveBool"
              type="checkbox"
              value="false"
            />
            <label
              class="form-check-label"
              for="formBasicCheckbox"
              title=""
            >
              add searched drugs to saved list
            </label>
          </div>
        </div>
        <small
          class="form-text text-muted"
          id="searchHelpBlock"
        >
          please separate multiple drug names by commas
        </small>
        <br />
        <button
          class="btn btn-lg btn-dark btn btn-primary"
          type="submit"
        >
          search
        </button>
        <p
          id="error-holder"
        />
      </form>
      <br />
    </div>
    <p
      id="error-holder"
    />
    <p>
      sorry, no results were found
    </p>
  </div>
</DocumentFragment>
`;

exports[`interactions page works as expected matches snapshot when loading 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        search for high severity interactions with a single drug, or get all interactions between multiple drugs.
      </h3>
      <form
        class=""
      >
        <label
          class="form-label"
          for="drugName"
        >
          search interactions:
        </label>
        <input
          class="form-control"
          name="drugName"
          placeholder="drug name(s)"
          value="adderall"
        />
        <br />
        <div
          class="form-group"
        >
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="formBasicCheckbox"
              name="saveBool"
              type="checkbox"
              value="false"
            />
            <label
              class="form-check-label"
              for="formBasicCheckbox"
              title=""
            >
              add searched drugs to saved list
            </label>
          </div>
        </div>
        <small
          class="form-text text-muted"
          id="searchHelpBlock"
        >
          please separate multiple drug names by commas
        </small>
        <br />
        <button
          class="btn btn-lg btn-dark btn btn-primary"
          type="submit"
        >
          search
        </button>
        <p
          id="error-holder"
        />
      </form>
      <br />
    </div>
    <div>
      <p>
        Loading all high-severity interactions with adderall...
      </p>
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
    <p>
      sorry, no results were found
    </p>
  </div>
</DocumentFragment>
`;
